src/devices/Is31fl3731/Is31Fl3731.cs

```diff
@@ -0,0 +1,278 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.I2c;
+using System.Threading;
+
+namespace Iot.Device.Display
+{
+    /// <summary>
+    /// Represents an IS31FL3731 LED Matrix driver
+    /// </summary>
+    // Datasheet: https://cdn-learn.adafruit.com/assets/assets/000/030/994/original/31FL3731.pdf
+    // Product: https://www.adafruit.com/product/2946
+    // Product: https://www.adafruit.com/product/2974
+    public abstract class Is31fl3731 : IDisposable
+    {
+        // Register addresses
+        // Command register
+        // Points pages one to nine
+        // table 2 in datasheet
+        private const byte COMMAND_REGISTER = 0xFD;
+        private const byte FUNCTION_REGISTER = 0x0B;
+
+        // Response register
+        // Writes data to pages one to eight
+        // table 3 in datasheet
+        private const byte LED_REGISTER = 0x0;
+        private const int LED_REGISTER_LENGTH = BLINK_REGISTER - LED_REGISTER;
+        private const byte BLINK_REGISTER = 0x12;
+        private const byte BLINK_REGISTER_LENGTH = PWM_REGISTER - BLINK_REGISTER;
+        private const byte PWM_REGISTER = 0x24;
+        private const byte PWM_REGISTER_END = 0xB3;
+        private const byte PWM_REGISTER_LENGTH = (PWM_REGISTER_END - PWM_REGISTER) + 1;
+
+        // Function register
+        // Writes data to page nine
+        // table 3 in datasheet
+        private const byte CONFIGURATION_REGISTER = 0x0;
+        private const byte DISPLAY_REGISTER = 0x5;
+        private const byte SHUTDOWN = 0x0A;
+
+        // Values
+        private readonly byte[] _disable_all_leds_data = new byte[LED_REGISTER_LENGTH];
+        private readonly byte[] _enable_all_leds_data = new byte[LED_REGISTER_LENGTH];
+
+        private I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// Initialize IS31FL3731 device
+        /// </summary>
+        /// <param name="i2cDevice">The <see cref="System.Device.I2c.I2cDevice"/> to create with.</param>
+        public Is31fl3731(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;
+            _enable_all_leds_data.AsSpan().Fill(0xff);
+        }
+
+        /// <summary>
+        /// Initialize IS31FL3731 device
+        /// </summary>
+        /// <param name="i2cDevice">The <see cref="System.Device.I2c.I2cDevice"/> to create with.</param>
+        /// <param name="width">The width of the LED matrix.</param>
+        /// <param name="height">The height of the LED matrix.</param>
+        public Is31fl3731(I2cDevice i2cDevice, int width, int height)
+        : this(i2cDevice)
+        {
+            Width = width;
+            Height = height;
+        }
+
+        /// <summary>
+        /// Default I2C address for device.
+        /// </summary>
+        public static readonly int DefaultI2cAddress;
```

Comment:
ok, now I understand why no the const in the other classes :-)

Reply:
Right.